scipy/stats/morestats.py

```diff
@@ -32,11 +33,20 @@ def __dir__():
 def __getattr__(name):
     if name not in __all__:
         raise AttributeError(
-            "scipy.stats.morestats is deprecated and has no attribute "
-            f"{name}. Try looking in scipy.stats instead.")
+            "`scipy.stats.morestats` is deprecated and has no attribute "
+            f"`{name}`.")
 
-    warnings.warn(f"Please use `{name}` from the `scipy.stats` namespace, "
-                  "the `scipy.stats.morestats` namespace is deprecated.",
-                  category=DeprecationWarning, stacklevel=2)
+    attr = getattr(scipy.stats, name, None)
```

Comment:
Choose the appropriate message depending on whether the attribute is public in `scipy.stats` or not.